zstacklib/zstacklib/utils/upload_task.py

```
@@ -0,0 +1,311 @@
+
+            slice_downloaded_size = 0
+            try:
+                reader = SizedReader(p.fp, None, param.slice_offset)
+                remaining = param.slice_size
+                bytes_read = 0
+                md5 = hashlib.md5()
+                chunks = []
+                chunk_size = 32 * 1024
+                while remaining > 0:
+                    if task.lastError:
+                        raise Exception(task.lastError)
```

Ruff (0.14.0)

Create your own exception

(TRY002)